shanoir-ng-studies/src/main/java/org/shanoir/ng/manufacturermodel/service/ManufacturerServiceImpl.java

```diff
@@ -32,25 +46,75 @@
 @Service
 public class ManufacturerServiceImpl implements ManufacturerService {
 
+	@Autowired
 	private ManufacturerRepository repository;
 
-	protected Manufacturer updateValues(Manufacturer from, Manufacturer to) {
-		return to;
+	@Autowired
+	private RabbitTemplate rabbitTemplate;
+
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	@Autowired
+	private AcquisitionEquipmentRepository acquisitionEquipmentRepository;
+	@Autowired
+	private ManufacturerModelRepository manufacturerModelRepository;
+
+	@Autowired
+	private StudyUserUpdateBroadcastService studyUserUpdateBroadcastService;
+
+	protected Manufacturer updateValues(Manufacturer manu, Manufacturer manuDb) {
+		manuDb.setName(manu.getName());
+
+		try {
+			updateManufacturer(manu);
+		} catch (MicroServiceCommunicationException e) {
+			throw new RuntimeException(e);
+		}
+		return manuDb;
 	}
 
+	public boolean updateManufacturer(Manufacturer manufacturer) throws MicroServiceCommunicationException {
+		try {
+			String manuName = manufacturer.getName();
+			if (manufacturer.getId() == null) {
+				return true;
+			}
+			List<ManufacturerModel> listManuModel = manufacturerModelRepository.findByManufacturerId(manufacturer.getId()).orElse(null);
+			if (listManuModel == null) {
```

Comment:
Does the method returns null or an empty list ? In that case, check with CollectionUtils.isempty ?